### PR TITLE
[sui-framework] Refactor `id` module to `object`

### PR DESCRIPTION
**Pull request by tnowacki**
- `sui::id` is now `sui::object`
- `VersionedID` renamed to `Info`
- Various ID and Info helpers have been renamed
- Moved object creation from tx_context to object

- A lot of changes to docs, but I will be doing a more detailed pass in a future PR

### CODE REVIEW

**Review by sblackshear — APPROVED** [review]
Need that ol' refactor tractor macro

File: doc/src/build/move.md [hunk 1]
```diff
@@ -161,9 +161,9 @@ and [structs](https://github.com/move-language/move/blob/main/language/documenta
 in the Move book.
 
 In order for a Move struct type to define a Sui object type such as
-`Coin`, its first field must be `id: VersionedID`, which is a
+`Coin`, its first field must be `info: Info`, which is a
 struct type defined in the
-[ID module](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/id.move). The
+[object module](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/object.move). The
```
Comment (Clay-Mysten) [inline doc/src/build/move.md:166]: Is this file being renamed?:
[object module](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/object.move)

It doesn't exist now.

**Review by Clay-Mysten — APPROVED** [review]
Great work, Todd! Thank you so much for including docs updates in this change. They LGTM assuming this linked file goes live with it (tried to pick through and confirm myself):
https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/object.move

Also, I think we should note this change here:
https://docs.sui.io/learn#see-whats-new

I will do that after you merge this and send the PR for your review. Thanks again!

File: doc/src/build/move.md (hunk 1 above)
Comment (Clay-Mysten) [reply doc/src/build/move.md:166]: > Correct, it's a renamed module after this PR lands

Hi Todd,

I missed a couple and am addressing them in:
https://github.com/MystenLabs/sui/pull/3300
https://github.com/MystenLabs/sui/pull/3301

Please take a look when you get a chance. Thanks!

**Pull request merged**